src/styles.test.ts

```diff
@@ -0,0 +1,213 @@
+  const text = figma.createText();
+  const rectangle = figma.createRectangle();
+  container.appendChild(text);
+  container.appendChild(rectangle);
+
+  container.gridStyleId = gridStyle.id;
+  text.textStyleId = textStyle.id;
+  rectangle.fillStyleId = paintStyle.id;
+  rectangle.effectStyleId = effectStyle.id;
+
+  const d = await dump([container], { styles: true });
+
+  const styleMap: F.StyleMap = {
+    [paintStyle.id]: {
+      key: paintStyle.key,
+      name: paintStyle.name,
+      styleType: paintStyle.type,
+      remote: paintStyle.remote,
+      description: paintStyle.description
+    },
+    [effectStyle.id]: {
+      key: effectStyle.key,
+      name: effectStyle.name,
+      styleType: effectStyle.type,
+      remote: effectStyle.remote,
+      description: effectStyle.description
+    },
+    [textStyle.id]: {
+      key: textStyle.key,
+      name: textStyle.name,
+      styleType: textStyle.type,
+      remote: textStyle.remote,
+      description: textStyle.description
+    },
+    [gridStyle.id]: {
+      key: gridStyle.key,
+      name: gridStyle.name,
+      styleType: gridStyle.type,
+      remote: gridStyle.remote,
+      description: gridStyle.description
+    }
+  };
+
+  expect(d.styles).toEqual(styleMap);
+});
+
+test("Doesn't include unused styles", async () => {
+  const { paintStyle } = createStyles();
+
+  // Intentionally only use paintStyle.
+  const container = figma.createFrame();
+  container.fillStyleId = paintStyle.id;
+
+  const d = await dump([container], { styles: true });
+
+  const styleMap: F.StyleMap = {
+    [paintStyle.id]: {
+      key: paintStyle.key,
+      name: paintStyle.name,
+      styleType: paintStyle.type,
+      remote: paintStyle.remote,
+      description: paintStyle.description
+    }
+  };
+
+  expect(d.styles).toEqual(styleMap);
+});
+
+test("Doesn't include same style multiple times", async () => {
+  const { paintStyle } = createStyles();
+
+  const container = figma.createFrame();
+  const rectangle = figma.createRectangle();
+  container.appendChild(rectangle);
+
+  // Use same style for both rectangle and container.
+  container.fillStyleId = paintStyle.id;
+  rectangle.fillStyleId = paintStyle.id;
+  // Use same style twice within rectangle (fill and stroke).
+  rectangle.strokeStyleId = paintStyle.id;
+
+  const d = await dump([container], { styles: true });
+
+  const styleMap: F.StyleMap = {
+    [paintStyle.id]: {
+      key: paintStyle.key,
+      name: paintStyle.name,
+      styleType: paintStyle.type,
+      remote: paintStyle.remote,
+      description: paintStyle.description
+    }
+  };
+
+  expect(d.styles).toEqual(styleMap);
+});
+
+test("Handles multiple styles of same type", async () => {
+  const { paintStyle } = createStyles();
+
+  const secondPaintStyle = figma.createPaintStyle();
+  // Hack because figma-api-stub doesn't create style keys.
+  (secondPaintStyle as { key: string }).key = getStyleKey(secondPaintStyle.id);
```

Comment:
Can you file an issue with `figma-api-stub` and link it here?

Reply:
https://github.com/react-figma/figma-api-stub/issues/61